### PR TITLE
Add ci:persist-prior-workflows tag to allow prior workflow to run until completion

### PR DESCRIPTION
**Pull request by sifive-benjamin-morse**
This is a CI only change that adds a new tag `ci:persist-prior-workflows`. When present, this tag will NOT run the `cancel-prior-workflows` job.

#### UI / API Impact
Developers will have a new tool that makes debugging CI issues easier.

#### Verilog / AGFI Compatibility
No change.

### Contributor Checklist
- [x] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [x] Did you add Scaladoc/docstring/doxygen to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous prints/debugging code?
- [x] Did you state the UI / API impact?
- [x] Did you specify the Verilog / AGFI compatibility impact?

- [ ] If applicable, did you regenerate and publicly share default AGFIs?

- [ ] If applicable, did you apply the `ci:fpga-deploy` label?

- [ ] If applicable, did you apply the `Please Backport` label?

### Reviewer Checklist (only modified by reviewer)
Note: to run CI on PRs from forks, comment `@Mergifyio copy main` and manage the change from the new PR.
- [ ] Is the title suitable for inclusion in the changelog and does the PR have a `changelog:<topic>` label?
- [ ] Did you mark the proper release milestone?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?

### CODE REVIEW

**Review by davidbiancolin — APPROVED** [review]
May as well announce this in firesim-dev.

**Comment by sifive-benjamin-morse** [comment]
> May as well announce this in firesim-dev.

I'll wait till Abe sees this, and then announce after the merge

**Review by abejgonzalez — APPROVED** [review]
LGTM. Out of curiosity what was the reason you wanted old workflows to stay up - to parallelize testing more?

**Comment by sifive-benjamin-morse** [comment]
> LGTM. Out of curiosity what was the reason you wanted old workflows to stay up - to parallelize testing more?

If the last step of CI is being flaky, parallel builds come in handy: As I continue to do builds to debug and resolve the intermittent failure, lets say I make a formatting mistake. I now would like to push my next commit to fix the formatting, but I don't want to cancel the previous run (because the more runs against a fix, the more confident I can be that I've fixed the intermittent failure)

**Pull request merged**
